interface/ceed-operator.c

```diff
@@ -791,6 +839,50 @@ int CeedOperatorGetFields(CeedOperator op, CeedInt *num_input_fields, CeedOperat
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Set the arbitrary points in each element for a CeedOperatorAtPoints.
+
+  Note: Calling this function asserts that setup is complete and sets the CeedOperator as immutable.
+
+  @param[in,out] op           CeedOperatorAtPoints
+  @param[in]     rstr_points  CeedElemRestriction for the coordinates of each point by element
+  @param[in]     point_coords CeedVector holding coordinates of each point
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Advanced
+**/
+int CeedOperatorAtPointsSetPoints(CeedOperator op, CeedElemRestriction rstr_points, CeedVector point_coords) {
+  CeedCheck(op->is_at_points, op->ceed, CEED_ERROR_MINOR, "Only defined for operator at points");
+  CeedCheck(!op->is_immutable, op->ceed, CEED_ERROR_MAJOR, "Operator cannot be changed after set as immutable");
+
+  CeedCall(CeedElemRestrictionReferenceCopy(rstr_points, &op->rstr_points));
+  CeedCall(CeedVectorReferenceCopy(point_coords, &op->point_coords));
+  return CEED_ERROR_SUCCESS;
```

Comment:
I feel like there needs to be some kind of check to make sure that the offsets arrays used in `CeedElemRestrictionCreateAtPoints` for the points input/output fields matches the ones provided here. Otherwise, I could imagine a scenario where we used an old version of the restriction for a field which would cause an opaque seg fault in the backend code. At the same time, that's kind of expensive, so maybe just check the first `num_elem + 1` entries (describing the number of points in each element).

Maybe we could implement this as a comparison function for `CeedElemRestriction`, then call it here?

Reply:
Yeah, I've been noodling on this and I agree. I think I have an approach (see future commit later this morning)

Reply:
alrighty, its the future now and I have added that check